### PR TITLE
Bringing back abstracted file sharing into kDrive

### PR DESCRIPTION
**Pull request by adrien-coye**
## Abstract

Bring back the `Core` work related to file sharing in this app.

Also abstracted the on the fly Heic -> Jpg ~[here](https://github.com/Infomaniak/ios-core/pull/53)~

~[use date as default name Core](https://github.com/Infomaniak/ios-core/pull/58)~
~[use date as default name CoreUI](https://github.com/Infomaniak/ios-core-ui/pull/11)~

Shared file name generation between manual import and PhotoLibraryUploader. 

Edited pictures in Photo.app have now a non coliding name.

## Details
- Speedup improvement, scale import workers dynamicaly given a device.
- chore: Bump Core / Core UI

### CODE REVIEW

File: kDriveCore/Utils/Files/FileImportHelper.swift [hunk 1]
```diff
@@ -198,141 +113,220 @@ public final class FileImportHelper {
...
+                            return await getText.result
+
+                        case .isUIImage:
+                            let getUIImage = try ItemProviderUIImageRepresentation(from: itemProvider)
+                            progress.addChild(getUIImage.progress, withPendingUnitCount: perItemUnitCount)
+                            return await getUIImage.result
+
+                        case .isImageData, .isCompressedData, .isMiscellaneous:
+
+                            // TODO: handle .heic .jpg tranform here.
```
Comment (adrien-coye) [inline kDriveCore/Utils/Files/FileImportHelper.swift:144]: TODO: handle .heic .jpg tranform here.

File: kDriveCore/Utils/Files/FileImportHelper.swift [hunk 2]
```diff
@@ -198,141 +113,220 @@ public final class FileImportHelper {
...
-                        } catch {
-                            DDLogError("could not move zipped file:\(error)")
-                            progress.completedUnitCount += perItemUnitCount
-                            errorCount += 1
-                            dispatchGroup.leave()
-                            return
-                        }
-                    }
+                    let fileName = url.lastPathComponent
+                    // TODO: pass the actual UTI
```
Comment (adrien-coye) [inline kDriveCore/Utils/Files/FileImportHelper.swift:175]: TODO: pass the actual UTI

File: kDriveCore/Utils/Files/FileImportHelper.swift [hunk 3]
```diff
@@ -198,141 +113,220 @@ public final class FileImportHelper {
...
-        dispatchGroup.notify(queue: .global()) {
-            completion(items, errorCount)
+            } catch {
+                DDLogError("[FileImportHelper] importItems:\(error)")
+            }
         }
 
         return progress
+
+        /*
```
Comment (adrien-coye) [inline kDriveCore/Utils/Files/FileImportHelper.swift:195]: TODO: Clean when done.

**Review by adrien-coye — COMMENTED** [review]
I did a quick self review.

File: kDrive/Utils/UploadCountManager.swift [hunk 4]
```diff
@@ -41,11 +46,19 @@ final class UploadCountManager {
     init(driveFileManager: DriveFileManager, didUploadCountChange: @escaping () -> Void) {
         self.driveFileManager = driveFileManager
         self.didUploadCountChange = didUploadCountChange
+        uploadCountObserver = uploadCountSubject
+            .debounce(for: .seconds(1), scheduler: DispatchQueue.main)
```
Comment (PhilippeWeidmann) [inline kDrive/Utils/UploadCountManager.swift:50]:
```suggestion
           .throttle(for: .seconds(1), scheduler: DispatchQueue.main, latest: true)
```
With debounce the value is never published

Comment (adrien-coye) [reply kDrive/Utils/UploadCountManager.swift:50]: Yep correct. nice catch 👍

**Review by PhilippeWeidmann — APPROVED** [review]

**Pull request merged**